WeakAuras/HelperFunctions.lua

```diff
@@ -0,0 +1,39 @@
+WeakAuras.helperFunction = {
+  -- Unit Aura functions that return info about the first Aura matching the spellName or spellID given on the unit. 
+  GetUnitAura = function(unit, spell, filter)
+      for i = 1, 40 do
+          local name, _, _, _, _, _, _, _, _, spellId = UnitAura(unit, i, filter)
+          if not name then return end
+          if spell == spellId or spell == name then
+              return UnitAura(unit, i, filter)
+          end
+      end
+  end
+  ,
+  GetUnitBuff = function(unit, spell, filter)
+      return GetUnitAura(unit, spell, filter)
+  end
+  ,
+  GetUnitDebuff = function(unit, spell, filter)
+      filter = filter and filter.."|HARMFUL" or "HARMFUL"
```

Comment:
Note that this is vulnerable to someone going `GetUnitDebuff("player", "some spell name",  "HELPFUL")`. I have no idea if this is worth guarding against, or if it should just be garbage in garbage out.